File: DAY 9/solutions.java/209. Minimum Size Subarray Sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {

        int left = 0;
        int windowSum = 0;
        int ans = Integer.MAX_VALUE;

        for (int right = 0; right < nums.length; right++) {

            windowSum += nums[right];

            while (windowSum >= target) {

                ans = Math.min(ans, right - left + 1);

                windowSum -= nums[left];
                left++;
            }
        }

        if (ans == Integer.MAX_VALUE) {
            return 0;
        }

        return ans;
    }
}

File: DAY 9/solutions.py/209. Minimum Size Subarray Sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:

        left = 0
        window_sum = 0
        ans = float('inf')

        for right in range(len(nums)):

            window_sum += nums[right]

            while window_sum >= target:

                ans = min(ans, right - left + 1)

                window_sum -= nums[left]
                left += 1

        if ans == float('inf'):
            return 0

        return ans